packages/vscode-dependencies-validation/test/vscodeMocks.ts

```diff
@@ -0,0 +1,189 @@
+    thisArg?: any
+  ): void {
+    throw new Error("Function not implemented.");
+  },
+  get: function (uri: Uri): readonly Diagnostic[] | undefined {
+    throw new Error("Function not implemented.");
+  },
+  has: function (uri: Uri): boolean {
+    throw new Error("Function not implemented.");
+  },
+  dispose: function (): void {
+    throw new Error("Function not implemented.");
+  },
+};
+
+export const textDocument: TextDocument = {
+  fileName: "",
+  isUntitled: false,
+  languageId: "",
+  version: 0,
+  isDirty: false,
+  isClosed: false,
+  save: function (): Thenable<boolean> {
+    throw new Error("Function not implemented.");
+  },
+  eol: 1,
+  lineCount: 0,
+
+  offsetAt: function (position: Position): number {
+    throw new Error("Function not implemented.");
+  },
+  positionAt: function (offset: number): Position {
+    throw new Error("Function not implemented.");
+  },
+  getText: function (range?: Range): string {
+    throw new Error("Function not implemented.");
+  },
+  getWordRangeAtPosition: function (
+    position: Position,
+    regex?: RegExp
+  ): Range | undefined {
+    throw new Error("Function not implemented.");
+  },
+  validateRange: function (range: Range): Range {
+    throw new Error("Function not implemented.");
+  },
+  validatePosition: function (position: Position): Position {
+    throw new Error("Function not implemented.");
+  },
+  // @ts-expect-error -- https://stackoverflow.com/questions/68799234/typescript-pick-only-specific-method-from-overload-to-be-passed-to-parameterst
+  lineAt: function (line: Position): TextLine {
+    throw new Error("Function not implemented.");
+  },
+};
+
+export const position: Position = {
+  line: 0,
+  character: 0,
+  isBefore: function (other: Position): boolean {
+    throw new Error("Function not implemented.");
+  },
+  isBeforeOrEqual: function (other: Position): boolean {
+    throw new Error("Function not implemented.");
+  },
+  isAfter: function (other: Position): boolean {
+    throw new Error("Function not implemented.");
+  },
+  isAfterOrEqual: function (other: Position): boolean {
+    throw new Error("Function not implemented.");
+  },
+  isEqual: function (other: Position): boolean {
+    throw new Error("Function not implemented.");
+  },
+  compareTo: function (other: Position): number {
+    throw new Error("Function not implemented.");
+  },
+  // @ts-expect-error -- https://stackoverflow.com/questions/68799234/typescript-pick-only-specific-method-from-overload-to-be-passed-to-parameterst
+  translate: function (lineDelta?: number, characterDelta?: number): Position {
+    throw new Error("Function not implemented.");
+  },
+  // @ts-expect-error -- https://stackoverflow.com/questions/68799234/typescript-pick-only-specific-method-from-overload-to-be-passed-to-parameterst
+  with: function (line?: number, character?: number): Position {
+    throw new Error("Function not implemented.");
+  },
+};
+
+export const range: Range = {
```

Comment:
you may be able a generic range implementation that fits or mostly fits this API